services/provider/server/consumer.go

```diff
@@ -74,8 +74,13 @@ func (c *ocsConsumerManager) Create(ctx context.Context, onboard ifaces.StorageC
 			},
 		},
 		Spec: ocsv1alpha1.StorageConsumerSpec{
-			Enable:            false,
-			StorageQuotaInGiB: storageQuotaInGiB,
+			Enable: false,
+			StorageQuotaInGiB: func() int {
+				if storageQuotaInGiB != nil {
+					return int(*storageQuotaInGiB)
```

Comment:
converting uint to int here, yes, we may not have larger values where we overflow but why not using `uint(*storageQuotaInGiB)`?

Reply:
ouch, I think there was some discussion which made this field in token struct as uint, seems non consistent with only a benefit of reducing one extra check for signedness.

Anyways, can't change that now.

Reply:
What is the issue with converting the unit to int? We are not expecting very large numbers here.
How does moving the conversion from outside the function into the function solves the problem?